File: app/src/main/java/fr/atech/robotcom/ui/configuration/ConfigurationFragment.java
package fr.atech.robotcom.ui.configuration;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProviders;

import fr.atech.robotcom.R;

public class ConfigurationFragment extends Fragment {

    private ConfigurationViewModel configurationViewModel;

    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container, Bundle savedInstanceState) {
        configurationViewModel =
                ViewModelProviders.of(this).get(ConfigurationViewModel.class);
        View root = inflater.inflate(R.layout.fragment_configuration, container, false);
        final TextView textView = root.findViewById(R.id.editText_config_ip_robot);
        configurationViewModel.getText().observe(getViewLifecycleOwner(), new Observer<String>() {
            @Override
            public void onChanged(@Nullable String s) {
                textView.setText(s);
            }
        });
        return root;
    }
}

File: app/src/main/java/fr/atech/robotcom/ui/configuration/ConfigurationFragment.kt
package fr.atech.robotcom.ui.configuration

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders
import fr.atech.robotcom.R

class ConfigurationFragment : Fragment() {

    private lateinit var configurationViewModel: ConfigurationViewModel

    override fun onCreateView(
            inflater: LayoutInflater,
            container: ViewGroup?,
            savedInstanceState: Bundle?
    ): View? {
        configurationViewModel =
                ViewModelProviders.of(this).get(ConfigurationViewModel::class.java)
        val root = inflater.inflate(R.layout.fragment_configuration, container, false)
        val textView: TextView = root.findViewById(R.id.text_configuration)
        configurationViewModel.text.observe(viewLifecycleOwner, Observer {
            textView.text = it
        })
        return root
    }
}
